protoc_plugin/lib/src/enum_generator.dart

```diff
@@ -175,11 +177,32 @@ class EnumGenerator extends ProtobufContainer {
       out.println('];');
       out.println();
 
-      out.println(
-          'static final $coreImportPrefix.Map<$coreImportPrefix.int, $classname> _byValue ='
-          ' $protobufImportPrefix.ProtobufEnum.initByValue(values);');
-      out.println('static $classname? valueOf($coreImportPrefix.int value) =>'
-          ' _byValue[value];');
+      var useList = _canonicalValues.isEmpty;
+      if (_canonicalValues.isNotEmpty) {
+        if (_canonicalValues.every((val) => !val.number.isNegative)) {
+          if (_canonicalValues.length / (_canonicalValues.last.number + 1) >=
```

Comment:
TODO: Can the largest known enum value be `-1`, or do we always need to define `0`? If we don't need to define `0` always then this can divide by zero when the largest enum value is `-1`.

Reply:
It looks like the first enum value needs to be `0`. Resolving.